### DISCUSSION TITLE
How to filter realtime subscription?

### DISCUSSION THREAD

**Discussion by laurentS** [supabase/supabase#546] [Questions] [answered]
I am looking for a way to do something like:
```javascript
const mySubscription = supabase
  .from('countries')
  .eq('country_type', 5)  // this does NOT work
  .on('*', payload => {
    console.log('Change received!', payload)
  })
  .subscribe()
```
basically, only get updates for a subset of rows in the table.
I tried
```javascript
const mySubscription = supabase
  .from('countries:country_type=eq.5')
  .on('*', handleRecordsChanged)
  .subscribe()
```
This seems to work for INSERT and UPDATE, but I don't get anything for DELETE (meaning if I perform these on the table).

Is this a bug or by design? If the latter, how can I subscribe to only get updates for the rows I'm interested in?

**Comment by kiwicopple** [comment, marked as answer]
Hey @laurentS - I thought this was in the docs, but I think it must have been removed. I added a note here:

https://supabase.io/docs/client/subscribe#listening-to-deletes

By default Supabase not send deleted records. If you want to receive the deleted record you can enable full replication for the table you are listening too: 

```sql
alter table "your_table" replica identity full;
```

**Reply by laurentS** [reply]
Thanks @kiwicopple! The `alter table` line you wrote seems to work with my second code snippet above, I'm now getting updates upon deletes!
